commit f545b6ae00ef77df710ddace328d13e8c5c265bf
Author: Sara Hartse <sara.hartse@gmail.com>
Date:   Fri Feb 15 14:44:56 2019 -0800

    Delay injection can cause indefinitely hung zios
    
    If we hit the (NSEC_TO_TICK(diff) == 0) condition in
    zio_delay_interrupt, zio_interrupt is never called and the
    zio does not progress.
    
    Reviewed by: Matt Ahrens <mahrens@delphix.com>
    Reviewed-by: Brian Behlendorf <behlendorf1@llnl.gov>
    Reviewed-by: George Melikov <mail@gmelikov.ru>
    Signed-off-by: sara hartse <sara.hartse@delphix.com>
    Closes #8404

Index: zfs/module/zfs/zio.c
===================================================================
--- zfs.orig/module/zfs/zio.c
+++ zfs/module/zfs/zio.c
@@ -1621,6 +1621,7 @@ zio_delay_interrupt(zio_t *zio)
 			if (NSEC_TO_TICK(diff) == 0) {
 				/* Our delay is less than a jiffy - just spin */
 				zfs_sleep_until(zio->io_target_timestamp);
+				zio_interrupt(zio);
 			} else {
 				/*
 				 * Use taskq_dispatch_delay() in the place of
